src/approaches/pg3_approach.py

```diff
@@ -0,0 +1,429 @@
+"""Policy-guided planning for generalized policy generation (PG3).
+
+Example command line:
+    python src/main.py --approach pg3 --seed 0 \
+        --env pddl_easy_delivery_procedural_tasks \
+        --strips_learner oracle --num_train_tasks 10
+"""
+from __future__ import annotations
+
+import abc
+import functools
+import logging
+from typing import Dict, FrozenSet, Iterator, List, Optional, Sequence, Set, \
+    Tuple
+from typing import Type as TypingType
+
+import dill as pkl
+from typing_extensions import TypeAlias
+
+from predicators.src import utils
+from predicators.src.approaches import ApproachFailure
+from predicators.src.approaches.nsrt_metacontroller_approach import \
+    NSRTMetacontrollerApproach
+from predicators.src.settings import CFG
+from predicators.src.structs import NSRT, Box, Dataset, GroundAtom, \
+    GroundAtomTrajectory, LDLRule, LiftedAtom, LiftedDecisionList, \
+    LowLevelTrajectory, ParameterizedOption, Predicate, State, Task, Type, \
+    Variable, _GroundNSRT
+
+
+class PG3Approach(NSRTMetacontrollerApproach):
```

Comment:
it's cool that this subclasses NSRTMetacontrollerApproach. that means we can try it with all envs, not just pddl envs, right? are there any conceptual blockades there?